### PR TITLE
ISPN-9000 Added email notify on failure

### PR DESCRIPTION
**Pull request by rigazilla**
This enables email notification when a build fails.

https://issues.jboss.org/browse/ISPN-9000

### CODE REVIEW

**Comment by tristantarrant** [comment]
Seems like you've removed a bunch of stuff :)

File: Jenkinsfile [hunk 1]
```diff
@@ -10,76 +10,17 @@ pipeline {
...
-        always {
-            // Archive logs and dump files
-            sh 'find . \\( -name "*.log" -o -name "*.dump*" -o -name "hs_err_*" \\) -exec xz {} \\;'
-            archiveArtifacts allowEmptyArchive: true, artifacts: '**/*.xz,documentation/target/generated-html/**'
-
-            // Clean
-            sh 'git clean -fdx -e "*.hprof" || echo "git clean failed, exit code $?"'
+       failure {
+            emailext to: '${DEFAULT_RECIPIENTS}', subject: '${DEFAULT_SUBJECT}', 
+            body: '${DEFAULT_CONTENT}'
```
Comment (slaskawi) [inline Jenkinsfile]: Is it possible to include a link to the job that failed? Or is it included automatically?

Comment (rigazilla) [reply Jenkinsfile]: @tristantarrant yeah I had removed the actual build to test just the email stuff. Now everything should in in place.

@slaskawi with the default subject/body (defined in jenkins/configuration) mail looks like this:

    Subject: CppRelease » Nightly CPP - Build # 62 - Failure!
    Body: CppRelease » Nightly CPP - Build # 62 - Failure: Check console output at https://ci.infinispan.org/job/CppRelease/job/Nightly%20CPP/62/ to view the results.

We can add more info if needed.

**Comment by rigazilla** [comment]
To enable the email notify the build must set 
    
      params.BUILD_RESULT_NOTIFY = 'EMAIL'

**Comment by rigazilla** [comment]
If enabled email will be sent if result status is:

failure, unstable
on the first success build

**Comment by danberindei** [comment]
@tristantarrant you mean in the build configuration, or in the Jenkinsfile itself?

**Comment by tristantarrant** [comment]
Ideally we should have this enabled for release branches and disabled for PRs.

**Comment by danberindei** [comment]
For PRs I see emailext has an option to send the email only to "culprits" (`recipientProviders: [culprits()]`). But I'm not sure how reliable the culprit detection is, it may require the commit email address and the Jenkins user's email address to match (mine doesn't).

File: Jenkinsfile [hunk 2]
```diff
@@ -81,5 +81,50 @@ pipeline {
             // Clean
             sh 'git clean -fdx -e "*.hprof" || echo "git clean failed, exit code $?"'
         }
+        changed {
```
Comment (slaskawi) [inline Jenkinsfile]: A line break before this one would be nice.

Comment (rigazilla) [reply Jenkinsfile]: ok

**Comment by rigazilla** [comment]
@slaskawi it's ok for me to merge this, I can enable the mail notify on the infinispan/branches pipelines and see the result (spam level and usefulness).
@danberindei if I understand correclty culprits are all the committers since the last successful build, but I would evaluate this option later, if we are unhappy with the defaults recipients list.

**Pull request merged**